### PR TITLE
Add MediaShare settings

### PR DESCRIPTION
**Pull request by gettinToasty**
very much a WIP, it looks like there isn't even an api endpoint for this yet.

Things to do:
* Add banned media component <-- Completed By Oat
* Change the security slider from displaying integers to displaying words
* Get actual descriptions and assets for Media Share in the source showcase

### CODE REVIEW

File: app/components/SourceSelector.vue.ts [hunk 1]
```diff
@@ -44,7 +45,7 @@ const sourceIconMap = {
   scene: 'far fa-object-group',
   color_source: 'fas fa-fill',
   openvr_capture: 'fab fa-simplybuilt fa-rotate-180',
-  liv_capture: 'fab fa-simplybuilt fa-rotate-180'
+  liv_capture: 'fab fa-simplybuilt fa-rotate-180',
```
Comment (gettinToasty) [inline app/components/SourceSelector.vue.ts]: just fyi i don't think trailing commas are a part of our style guide

File: app/components/widgets/WidgetSettings.vue.ts [hunk 2]
```diff
@@ -98,6 +98,7 @@ export default class WidgetSettings<TData extends IWidgetData, TService extends
 
     this.requestState = 'pending';
 
+
```
Comment (gettinToasty) [inline app/components/widgets/WidgetSettings.vue.ts]: extra space

File: app/services-manager.ts [hunk 3]
```diff
@@ -181,7 +182,8 @@ export class ServicesManager extends Service {
     TipJarService,
     SponsorBannerService,
     MediaGalleryService,
-    AnnouncementsService
+    AnnouncementsService,
+    MediaShareService,
```
Comment (gettinToasty) [inline app/services-manager.ts]: trailing comma

File: app/services/sources/sources-api.ts [hunk 4]
```diff
@@ -97,7 +97,8 @@ export type TSourceType =
   'scene' |
   'ndi_source' |
   'openvr_capture' |
-  'liv_capture'
+  'liv_capture' |
+  'media_share'
```
Comment (gettinToasty) [inline app/services/sources/sources-api.ts]: is media_share a source type that isn't a browser source?

**Comment by Nitsorn** [comment]
@gettinToasty fixed all your comments

**Comment by gettinToasty** [comment]
AFAIK this is complete but i don't want to review it myself since i wrote a bunch of the code

**Comment by Nitsorn** [comment]
`new-chatbot` branch contains all of this code and is merged. We can close this.

**Pull request closed without merging**